buildroot/bin/build_all_examples

```diff
@@ -35,16 +35,17 @@ else
   echo "Resuming from $2"
 fi
 
-TMPDIR=`mktemp -d`
+TMPDIR="$PWD/tmp"
```

Comment:
I prefer to use a temporary folder in a standard location rather than put one into my workspace. Configurations aren't too large, and the system should purge any orphaned temporary directories when it runs its maintenance tasks.

Reply:
If the temp folder should be reused, what I have done in the past was write a little dot file inside the `./.pio` folder that contains the old information. The `mftest` script does this so that `mftest -r` knows what the last environment was. Another way to do it would be to do an `export $TMPDIR` to the shell environment, and then look for it when the script starts.

Reply:
Not sure what's better. I've called it "tmp", but it could also be called "build" and the intention would be more clear. Another possibility would be to add a symbolic link to the mktemp'ed folder named `.pio/example` and check for it upon entering the script. But we'd need something to clear it or the link will linger to an invalid path. I'm not fan of scripts polluting the environment.

Reply:
I don't think it's intrusive to add a tiny hidden file to the `.pio` folder. And it's the only solution allowed to remember prior state because it's not possible to update the current environment from a script unless it is initiated with `source`, but in that case any `exit` ends up quitting the top shell.